### PR TITLE
SNOW-52: Adding Markdown widget type

### PR DESCRIPTION
**Pull request by VTamas1**
SNOW-52

### CODE REVIEW

**Review by Luko6 — COMMENTED** [review]
You have to create an OSOCE PR if you change the open source submodules.

**Comment by wAsnk** [comment]
> You have to create an OSOCE PR if you change the open source submodules.

And what changes should the PR contain?

**Comment by sarahelsaig** [comment]
> And what changes should the PR contain?

Create a new branch from OSOCE's `dev` called `issue/SNOW-52` and update each submodule that exists in both and you have updated here to their respective `issue/SNOW-52` branches. So HL and HE for this issue. This is for very basic regression testing on OSOCE, and to make sure all the submodules in that repo are up-to-date the same way you update the submodules here.

By the way, you don't have PRs for your HelpfulExtensions and BaseTheme branches.

**Comment by wAsnk** [comment]
> > And what changes should the PR contain?
> 
> Create a new branch from OSOCE's `dev` called `issue/SNOW-52` and update each submodule that exists in both and you have updated here to their respective `issue/SNOW-52` branches. So HL and HE for this issue. This is for very basic regression testing on OSOCE, and to make sure all the submodules in that repo are up-to-date the same way you update the submodules here.
> 
> By the way, you don't have PRs for your HelpfulExtensions and BaseTheme branches.

Thanks, tho this is the HelpfulExtensions PR. The BaseTheme shouldn't be changed, @VTamas1 should remove that.
edit: You might think about HelpfulLibraries. But that should be removed aswell, there are no changes, just a simple dev => issue/SNOW-52 merge.

@VTamas1 Please checkout to the correct branch in BaseTheme and HelpfulLibraries submodule, then push in your branch in the main repo.

File: Extensions/Widgets/Migrations.cs.orig [hunk 1]
```diff
@@ -0,0 +1,113 @@
+using OrchardCore.ContentManagement.Metadata;
```
Comment (Luko6) [inline Extensions/Widgets/Migrations.cs.orig]: delete .orig file

File: Extensions/Widgets/WidgetTypes.cs.orig [hunk 2]
```diff
@@ -0,0 +1,14 @@
+namespace Lombiq.HelpfulExtensions.Extensions.Widgets
```
Comment (Luko6) [inline Extensions/Widgets/WidgetTypes.cs.orig]: delete .orig file

File: Extensions/Widgets/Migrations.cs [hunk 3]
```diff
@@ -57,7 +65,7 @@ public int UpdateFrom1()
             .WithPart("FlowPart", part => part.WithPosition("1"))
         );
 
-        return 2;
+        return 3;
```
Comment (Luko6) [inline Extensions/Widgets/Migrations.cs]: Revert this change

File: Extensions/Widgets/Migrations.cs [hunk 4]
```diff
@@ -67,6 +75,19 @@ public int UpdateFrom2()
             .Stereotype(CommonStereotypes.Widget)
         );
 
-        return 3;
+        return 4;
```
Comment (Luko6) [inline Extensions/Widgets/Migrations.cs]: Revert this change

**Review by Luko6 — CHANGES_REQUESTED** [review]

File: Extensions/Widgets/Migrations.cs.orig (hunk 1 above)
Comment (sarahelsaig) [reply Extensions/Widgets/Migrations.cs.orig]: Also add `*.orig` to the _.gitignore_ file to prevent this from the future.

**Review by Luko6 — APPROVED** [review]

**Pull request merged**